src/components/Api.js

```diff
@@ -0,0 +1,96 @@
+class Api {
+    constructor({ baseUrl, headers }) {
+      this._baseUrl = baseUrl;
+      this._headers = headers;
+    }
+  
+    _checkResponse(res) {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    }
+  
+    getUserInfo() {
+      return fetch(`${this._baseUrl}/users/me`, { headers: this._headers }).then(
+        this._checkResponse
+      );
+    }
+  
+    getCardsList() {
+      return fetch(`${this._baseUrl}/cards`, { headers: this._headers }).then(
+        this._checkResponse
+      );
+    }
+  
+    updateUserInfo(name, about) {
+      return fetch(`${this._baseUrl}/users/me`, {
+        method: "PATCH",
+        headers: this._headers,
+        body: JSON.stringify({
+          name,
+          about,
+        }),
+      }).then(this._checkResponse);
+    }
+  
+    addCard(name, link) {
+      return fetch(`${this._baseUrl}/cards`, {
+        method: "POST",
+        headers: this._headers,
+        body: JSON.stringify({
+          name,
+          link,
+        }),
+      }).then(this._checkResponse);
+    }
+  
+    deleteCard(cardId) {
+      return fetch(`${this._baseUrl}/cards/${cardId}`, {
+        method: "DELETE",
+        headers: this._headers,
+      }).then((res) => {
```

Comment:
There should not be code duplications. You should use `this._checkResponse` method

Reply:
Do you mean I should removed the whole code?

Reply:
No, you should reuse `this._checkResponse` in `then` block as you do that in other methods